### PR TITLE
test(agent): add agent loop conversion test suite

### LINKED ISSUE #944: Enhancement: Add Unit Tests for Agent Loop Implementations
Enhancement: Add Unit Tests for Agent Loop Implementations
Problem

- Currently, agent loop implementations (e.g., uitars.py, anthropic.py, qwen.py) lack dedicated unit tests. This makes it difficult to:
- Verify message formatting logic without full integration tests
- Catch regressions when modifying agent behavior
- Validate fixes for serialization or API compatibility issues
- Contribute confidently to agent loop code

Proposed Solution
Add unit test coverage for agent loop modules in libs/python/agent/tests/, particularly:

High Priority

- test_uitars.py - Test message construction, ordering, and merging logic
- test_anthropic.py - Test Claude-specific formatting
- test_message_conversion.py - Test conversion functions across all loops

Test Coverage Should Include

- Message Formatting: Verify correct role ordering (system → user → assistant)
- Message Merging: Test consecutive message handling
- Edge Cases: Empty history, single messages, complex multi-turn scenarios
- Serialization: Validate output matches API expectations

Benefits

-  Faster development cycles (no need for full integration tests)
-  Easier to verify bug fixes (like recent LiteLLM serialization fix)
-  Better code quality and regression prevention
-  Lower barrier for new contributors

Example Test Structure
```
def test_uitars_message_ordering():
    """Verify messages are ordered: System -> History -> Current"""
    messages = [...]
    result = convert_uitars_messages_to_litellm(messages)
    assert result[0]["role"] == "system"
    assert result[-1]["role"] == "user"
```
Implementation Notes

- Use pytest (already in test infrastructure)
- Mock litellm and model-specific dependencies
- Focus on logic testing, not API integration
- Keep tests fast and isolated

Related Work
**This enhancement was identified while fixing #[ISSUE_NUMBER] (UI-TARS serialization error), where testing the fix required manual simulation scripts instead of proper unit tests.**

### PR DESCRIPTION
**Pull request by BloggerBust**
- Add unit tests for Anthropic, shared message conversion utilities, and UITARS
- Drop user messages that contain only filtered content to avoid invalid API payloads
- Guard computer_call_output handling for non dict outputs
- Narrow agent Python version constraint and add optional test dependencies

Closes #944

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed message handling to properly drop empty filtered image content
  * Enhanced type validation for tool outputs

* **Tests**
  * Added comprehensive test coverage for message conversion workflows
  * Added extensive tests for anthropic and UITARS integration

* **Chores**
  * Updated test dependencies and configuration

### CODE REVIEW

**Comment by BloggerBust** [comment]
Just checking in! I am happy to revise if needed or adjust scope. Let me know if there is anything blocking review.

File: pyproject.toml [hunk 1]
```diff
@@ -53,12 +53,9 @@ docs-scripts = [
...
 ]
 test = [
-    "aioresponses>=0.7.4",
+    "pytest>=8.0.0",
     "pytest-asyncio>=0.21.1",
-    "pytest-cov>=4.1.0",
     "pytest-mock>=3.10.0",
-    "pytest-xdist>=3.6.1",
-    "pytest>=8.0.0",
 ]
```
[the remark this answers is not in the document]
Comment (BloggerBust) [reply pyproject.toml:62]: Thanks for catching this. The dependencies have been restored.
